src/_pytest/assertion/rewrite.py

```diff
@@ -703,24 +703,15 @@ def run(self, mod: ast.Module) -> None:
             return
         pos = 0
         for item in mod.body:
-            if (
-                expect_docstring
-                and isinstance(item, ast.Expr)
-                and isinstance(item.value, ast.Constant)
-                and isinstance(item.value.value, str)
-            ):
-                doc = item.value.value
-                if self.is_rewrite_disabled(doc):
-                    return
-                expect_docstring = False
-            elif (
-                isinstance(item, ast.ImportFrom)
-                and item.level == 0
-                and item.module == "__future__"
-            ):
-                pass
-            else:
-                break
+            match item:
```

Comment:
this looks interesting - whats the performance impact - i would expect match to be faster but that may be wishful thinking

Reply:
Unfortunately we had mixed results in pylint regarding performance. What do you suggest as a benchmark ?

Reply:
We need code fragments that trigger the different cases and then we need a codegen that given a number creates that many instances 

Then we can observe/profile 

Im not sure whether we should synthesize a ast or just text

Reply:
Seems worse than i hoped

Reply:
The examples I took where specifically the one where the diff was very favorable (There's a lot of match on raw string that could be done but the performance is worse in this case as there's no index lookup creation), so I also expected better. From what I read it should be a neutral change performance wise except if isinstance were smartly grouped together with ``and`` / ``or`` to do less checks. Maybe the regression is due to microbenchmarking, maybe match is only a readability change. It's hard to find information that is not slop about this topic. (found this for example : https://discuss.python.org/t/pattern-matching-optimization-comparison-of-values-specified-through/20791)

Reply:
Marc Mueller dug into the CPython implementation here (pylint specific but still relevant) : https://github.com/pylint-dev/pylint/pull/10544#issuecomment-3289962235

Reply:
Opened an issue for CPython: https://github.com/python/cpython/issues/138912